### DISCUSSION TITLE
Client side bundles has  most of  server side code

### DISCUSSION THREAD

**Discussion by udaykumarelluri** [vercel/next.js#19144] [Help] [answered]
In client side bundles has most of server side code which is not being used in client side how can i remove those 

i used this 
 if (typeof window === "undefined") {}

it is working webpack reconize the above line and will remove 

Has any other option ? to remove the server side code in the client bundles

**Comment by jamesmosier** [comment, marked as answer]
What code in particular? 

`typeof window === 'undefined'` should ensure code is not in client bundles.

But if you use imports in a file those are only eliminated when their usage is inside `getServerSideProps`, `getStaticProps`, and `getStaticPaths` as shown here https://next-code-elimination.now.sh/
